src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,162 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int RESIZE_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int tableIndex = hash(key);
+        if (table[tableIndex] != null) {
+            Node<K, V> temp = table[tableIndex];
+            Node<K, V> nextNode;
+            if (Objects.equals(table[tableIndex].key, key)) {
+                table[tableIndex] = newNode;
+                table[tableIndex].next = temp.next;
+                return;
+            }
+            while (temp.next != null) {
+                if (hash(table[tableIndex].key) == hash(key)) {
+                    if (Objects.equals(temp.next.key, key)) {
+                        nextNode = temp.next;
+                        temp.next = newNode;
+                        newNode.next = nextNode;
+                        return;
+                    }
+                    temp = temp.next;
+                }
+            }
+            temp.next = newNode;
+        } else {
+            table[tableIndex] = newNode;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int tableIndex = hash(key);
+        if (table[tableIndex] == null) {
+            return null;
+        }
+        if (table[tableIndex].next != null) {
+            Node<K, V> temp = table[tableIndex];
+            while (temp != null && !Objects.equals(temp.key, key)) {
+                temp = temp.next;
+            }
+            if (temp == null) {
+                return null;
+            }
+            return temp.value;
+        }
+        return table[tableIndex].getValue();
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : (key.hashCode() & 0x7FFFFFFF) % table.length;
+    }
+
+    private void resize() {
+        final Node<K, V>[] oldTable = table;
+        int capacity = table.length;
+        capacity *= RESIZE_FACTOR;
+        table = (Node<K, V>[]) new Node[capacity];
+        threshold = (int) (capacity * LOAD_FACTOR);
+        size = 0;
+        for (int i = 0; i < oldTable.length; i++) {
+            Node<K, V> temp = oldTable[i];
+            while (temp != null) {
+                put(temp.key, temp.value);
+                temp = temp.next;
+            }
+        }
+    }
+
+    class Node<K, V> {
```

Comment:
Remember about access modifiers. All fields and methods should have access modifiers. The fields in the Node class should be private to adhere to the encapsulation principle.